connectors/sources/tests/test_confluence.py

```diff
@@ -5,16 +5,173 @@
+            "space": {"name": "DEMO"},
+            "_links": {
+                "webui": "/spaces/~1234abc/pages/4779/ES-scrum",
+            },
+        }
+    ],
+    "start": 0,
+    "limit": 1,
+    "size": 1,
+    "_links": {},
+}
+
+EXPECTED_PAGE = {
+    "_id": 4779,
+    "type": "page",
+    "_timestamp": "2023-01-24T04:07:19.672Z",
+    "title": "ES-scrum",
+    "body": "This is a test page",
+    "space": "DEMO",
+    "url": f"{HOST_URL}/spaces/~1234abc/pages/4779/ES-scrum",
+}
+
+EXPECTED_SPACE = {
+    "_id": 4554779,
+    "type": "Space",
+    "title": "DEMO",
+    "_timestamp": "2023-01-24T04:07:19+00:00",
+    "url": f"{HOST_URL}/spaces/DM",
+}
+
+RESPONSE_ATTACHMENT = {
+    "results": [
+        {
+            "id": "att3637249",
+            "title": "demo.py",
+            "type": "attachment",
+            "version": {"when": "2023-01-03T09:24:50.633Z"},
+            "extensions": {"fileSize": 230},
+            "_links": {
+                "download": "/download/attachments/1113/demo.py?version=1&modificationDate=1672737890633&cacheVersion=1&api=v2",
+                "webui": "/pages/viewpageattachments.action?pageId=1113&preview=demo.py",
+            },
+        }
+    ],
+    "start": 0,
+    "limit": 1,
+    "size": 1,
+    "_links": {},
+}
+
+EXPECTED_ATTACHMENT = {
+    "_id": "att3637249",
+    "type": "attachment",
+    "_timestamp": "2023-01-03T09:24:50.633Z",
+    "title": "demo.py",
+    "size": 230,
+    "space": "DEMO",
+    "page": "ES-scrum",
+    "url": f"{HOST_URL}/pages/viewpageattachments.action?pageId=1113&preview=demo.py",
+}
+
+RESPONSE_CONTENT = "# This is the dummy file"
+
+EXPECTED_CONTENT = {
+    "_id": "att3637249",
+    "_timestamp": "2023-01-03T09:24:50.633Z",
+    "_attachment": "IyBUaGlzIGlzIHRoZSBkdW1teSBmaWxl",
+}
+
+EXPECTED_BLOG = {
+    "_id": 4779,
+    "type": "blogpost",
+    "_timestamp": "2023-01-24T04:07:19.672Z",
+    "title": "demo-blog",
+    "body": "This is a test blog",
+    "space": "DEMO",
+    "url": f"{HOST_URL}/spaces/~1234abc/blogposts/4779/demo-blog",
+}
+
+EXPECTED_BLOG_ATTACHMENT = {
+    "_id": "att3637249",
+    "type": "attachment",
+    "_timestamp": "2023-01-03T09:24:50.633Z",
+    "title": "demo.py",
+    "size": 230,
+    "space": "DEMO",
+    "blog": "demo-blog",
+    "url": f"{HOST_URL}/pages/viewpageattachments.action?pageId=1113&preview=demo.py",
+}
+
+
+class MockJsonResponse:
+    """Class to mock json response of aiohttp session.get method"""
+
+    def __init__(self, json, status):
+        """Setup a json response"""
+        self._json = json
+        self.status = status
+
+    async def json(self):
+        """Method to return a JSON content of the result"""
+        return self._json
+
+    async def __aexit__(self, exc_type, exc, tb):
+        """Closes an async with block"""
+        pass
+
+    async def __aenter__(self):
+        """Enters an async with block"""
+        return self
+
+
+class MockObjectResponse:
```

Comment:
Could we use `AsyncMock` instead?

Reply:
Similarly here we need a StreamReader object to be returned named `content`, so we had to create a custom Class